lib/manageiq/api/common/rbac/acl.rb

```diff
@@ -0,0 +1,74 @@
+module ManageIQ
+  module API
+    module Common
+      module RBAC
+        class ACL
+          def create(resource_id, permissions)
+            permissions.collect do |permission|
+              create_acl(permission, resource_id)
+            end
+          end
+
+          def remove(acls, resource_id, permissions)
+            permissions.each_with_object(acls) do |permission, as|
+              delete_matching(as, resource_id, permission)
+            end
+          end
+
+          def add(acls, resource_id, permissions)
+            new_acls = permissions.each_with_object([]) do |permission, as|
+              next if find_matching(acls, resource_id, permission)
+
+              as << create_acl(permission, resource_id)
+            end
+            new_acls.empty? ? acls : new_acls + acls
```

Comment:
If it's empty, then the `+` operation will work as expected, so this line can just be `new_acls + acls`